### PR TITLE
Support uuid id type

### PR DESCRIPTION
**Pull request by javeme**
implement #617

Change-Id: Ie05d7a3609dd1695728f0a2f4a322448ac8aabeb

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java [hunk 1]
```diff
@@ -318,6 +326,7 @@ public BytesBuffer writeId(Id id, boolean big) {
                 len -= 1;
                 int high = len >> 8;
                 int low = len & 0xff;
+                assert high != 0x7f; // The tail 7 bits 1 reserved for UUID
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java]: 'len <= BIG_ID_MAX_LEN' can't ensure 'high != 0x7f',

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java [hunk 2]
```diff
@@ -45,8 +46,8 @@
     public static final int UINT16_MAX = ((short) -1) & 0xffff;
     public static final long UINT32_MAX = (-1) & 0xffffffffL;
 
-    public static final int ID_MAX_LEN = UINT8_MAX & 0x7f + 1; // 128
-    public static final int BIG_ID_MAX_LEN = UINT16_MAX & 0x7fff + 1; // 32768
+    public static final int ID_MAX_LEN = UINT8_MAX & 0x7f; // 127
+    public static final int BIG_ID_MAX_LEN = UINT16_MAX & 0x7fff; // 32767
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java]: max length is 32767-256

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java (hunk 1 above)
Comment (javeme) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java]: id len must be in range [0, 126]

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java (hunk 2 above)
Comment (javeme) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java]: yes, 1byte id len must be in range [0, fe],  2byte id len must be in range [0, feff]

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/id/Id.java [hunk 3]
```diff
@@ -31,5 +33,24 @@
...
+    public default boolean uuid() {
+        return this.type() == IdType.UUID;
+    }
+
+    public default boolean string() {
+        return this.type() == IdType.STRING;
+    }
+
+    public enum IdType {
+        UNKONW,
```
Comment (Linary) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/id/Id.java]: spell error UNKONW  -> UNKNOWN

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/id/IdGenerator.java [hunk 4]
```diff
@@ -32,12 +39,52 @@ public final static Id of(String id) {
...
+        }
+    }
+
+    public final static Id ofStoredString(String id, IdType type) {
+        switch (type) {
+            case LONG:
+                return of(LongEncoding.decode(id));
+            case UUID:
+                byte[] bytes = Base64.getDecoder().decode(id);
+                return of(bytes, type);
```
Comment (Linary) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/id/IdGenerator.java]: just return of(bytes, UUID);

**Pull request merged**